zerver/tests/test_home.py

```diff
@@ -736,83 +746,100 @@ def test_show_billing(self) -> None:
         user.role = UserProfile.ROLE_MEMBER
         user.is_billing_admin = True
         user.save(update_fields=["role", "is_billing_admin"])
         CustomerPlan.objects.all().delete()
-        self.assertNotInHomePage("Billing")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # billing admin, with sponsorship pending -> show billing link
+        # billing admin, with sponsorship pending and relam plan_type SELF_HOSTED -> show only billing link
         customer.sponsorship_pending = True
         customer.save(update_fields=["sponsorship_pending"])
-        self.assertInHomePage("Billing")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertTrue(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # billing admin, no customer object -> make sure it doesn't crash
+        # billing admin, no customer object and relam plan_type SELF_HOSTED -> neither billing link or plans
         customer.delete()
-        result = self._get_home_page()
-        self.check_rendered_logged_in_app(result)
-
-    def test_show_plans(self) -> None:
-        realm = get_realm("zulip")
-
-        # Don't show plans to guest users
-        self.login("polonius")
-        do_change_plan_type(realm, Realm.LIMITED, acting_user=None)
-        self.assertNotInHomePage("Plans")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # Show plans link to all other users if plan_type is LIMITED
-        self.login("hamlet")
-        self.assertInHomePage("Plans")
-
-        # Show plans link to no one, including admins, if SELF_HOSTED or STANDARD
-        do_change_plan_type(realm, Realm.SELF_HOSTED, acting_user=None)
-        self.assertNotInHomePage("Plans")
-
-        do_change_plan_type(realm, Realm.STANDARD, acting_user=None)
-        self.assertNotInHomePage("Plans")
-
-    def test_show_support_zulip(self) -> None:
+    def test_promote_sponsoring_zulip_in_realm(self) -> None:
         realm = get_realm("zulip")
 
-        self.login("hamlet")
-
-        self.assertInHomePage("Support Zulip")
-
         do_change_plan_type(realm, Realm.STANDARD_FREE, acting_user=None)
-        self.assertInHomePage("Support Zulip")
+        promote_zulip = promote_sponsoring_zulip_in_realm(realm)
+        self.assertTrue(promote_zulip)
 
-        with self.settings(PROMOTE_SPONSORING_ZULIP=False):
-            self.assertNotInHomePage("Support Zulip")
+        do_change_plan_type(realm, Realm.STANDARD_FREE, acting_user=None)
+        promote_zulip = promote_sponsoring_zulip_in_realm(realm)
+        self.assertTrue(promote_zulip)
```

Comment:
This seems to have two copies of this first test.

Reply:
> I think you lost the with statement check or something.

It's there at the [end](https://github.com/zulip/zulip/pull/18829/files#diff-e2d3dbded9826269a6d62f68ebd6031795e2996a2644372dc9568bf7282939caR840).